Build failure: vorbisgain-0.37

## Steps To Reproduce

Steps to reproduce the behavior:

1. Add package dependent on vorbisgain-0.37 (in this case, flacon and Qt5 soundkonverter).
2. Rebuild system.

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/gijgwgjb22bza79pfs41vfpkd7k1xbw2-vorbisgain-0.37.tar.gz
source root is vorbisgain-0.37
setting SOURCE_DATE_EPOCH to timestamp 1121381911 of file vorbisgain-0.37/README
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
mode of 'configure' changed from 0754 (rwxr-xr--) to 0755 (rwxr-xr-x)
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --disable-dependency-tracking --prefix=/nix/store/qs8zip2w65gqmzvmyn217z1l5dfh2bbk-vorbisgain-0.37 --mandir=/nix/store/qs8zip2w65gqmzvmyn217z1l5dfh2bbk-vorbisgain-0.37/share/man
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking for Ogg... yes
checking for Vorbis... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking how to run the C preprocessor... gcc -E
checking for egrep... grep -E
checking for ANSI C header files... no
checking whether termios.h defines TIOCGWINSZ... no
checking whether sys/ioctl.h defines TIOCGWINSZ... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking errno.h usability... yes
checking errno.h presence... yes
checking for errno.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for unistd.h... (cached) yes
checking utime.h usability... yes
checking utime.h presence... yes
checking for utime.h... yes
checking termios.h usability... yes
checking termios.h presence... yes
checking for termios.h... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for size_t... yes
checking whether lstat dereferences a symlink specified with a trailing slash... yes
checking whether stat accepts an empty string... no
checking for vprintf... yes
checking for _doprnt... no
checking for floor... no
checking for getcwd... yes
checking for memmove... yes
checking for memset... yes
checking for pow... no
checking for strchr... yes
checking for strdup... yes
checking for strerror... yes
checking for ov_read_float in -lvorbisfile... yes
checking if ov_read_float() takes four arguments... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating config.h
config.status: executing depfiles commands
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
make  all-am
make[1]: Entering directory '/build/vorbisgain-0.37'
gcc -DHAVE_CONFIG_H -I. -I. -I.     -O2 -fsigned-char  -c `test -f 'vorbisgain.c' || echo './'`vorbisgain.c
gcc -DHAVE_CONFIG_H -I. -I. -I.     -O2 -fsigned-char  -c `test -f 'vorbis.c' || echo './'`vorbis.c
gcc -DHAVE_CONFIG_H -I. -I. -I.     -O2 -fsigned-char  -c `test -f 'recurse.c' || echo './'`recurse.c
gcc -DHAVE_CONFIG_H -I. -I. -I.     -O2 -fsigned-char  -c `test -f 'misc.c' || echo './'`misc.c
[01m[Kmisc.c:[m[K In function '[01m[Kget_console_size[m[K':
[01m[Kmisc.c:268:9:[m[K [01;31m[Kerror: [m[Kimplicit declaration of function '[01m[Kisatty[m[K' [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration]8;;[m[K]
  268 |     if ([01;31m[Kisatty[m[K(fd))
      |         [01;31m[K^~~~~~[m[K
make[1]: *** [Makefile:256: misc.o] Error 1
make[1]: Leaving directory '/build/vorbisgain-0.37'
make: *** [Makefile:167: all] Error 2

```

</details>

## Additional context

<!-- Add any other context about the problem here. -->
First issue opened here. Let me know if I missed anything. 

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 25.05 (Warbler), 25.05.20241209.a73246e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/store/yw6kg4rb9v8s3ypjbpspig5r81m4lr5s-source`


## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@infinisil 


---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

@pSub appears to be the maintainer rather than @infinisil named on the ticket.
 
My first build after the Christmas break results in reproducing the build error:

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.64, NixOS, 25.05 (Warbler), 25.05.20241217.d3c42f1`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/22r7q7s9552gn1vpjigkbhfgcvhsrz68-source`
```

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/gijgwgjb22bza79pfs41vfpkd7k1xbw2-vorbisgain-0.37.tar.gz
source root is vorbisgain-0.37
setting SOURCE_DATE_EPOCH to timestamp 1121381911 of file vorbisgain-0.37/README
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
mode of 'configure' changed from 0754 (rwxr-xr--) to 0755 (rwxr-xr-x)
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --disable-dependency-tracking --prefix=/nix/store/qs8zip2w65gqmzvmyn217z1l5dfh2bbk-vorbisgain-0.37 --mandir=/nix/store/qs8zip2w65gqmzvmyn217z1l5dfh2bbk-vorbisgain-0.37/share/man
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking for Ogg... yes
checking for Vorbis... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking how to run the C preprocessor... gcc -E
checking for egrep... grep -E
checking for ANSI C header files... no
checking whether termios.h defines TIOCGWINSZ... no
checking whether sys/ioctl.h defines TIOCGWINSZ... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking errno.h usability... yes
checking errno.h presence... yes
checking for errno.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for unistd.h... (cached) yes
checking utime.h usability... yes
checking utime.h presence... yes
checking for utime.h... yes
checking termios.h usability... yes
checking termios.h presence... yes
checking for termios.h... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for size_t... yes
checking whether lstat dereferences a symlink specified with a trailing slash... yes
checking whether stat accepts an empty string... no
checking for vprintf... yes
checking for _doprnt... no
checking for floor... no
checking for getcwd... yes
checking for memmove... yes
checking for memset... yes
checking for pow... no
checking for strchr... yes
checking for strdup... yes
checking for strerror... yes
checking for ov_read_float in -lvorbisfile... yes
checking if ov_read_float() takes four arguments... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating config.h
config.status: executing depfiles commands
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
make  all-am
make[1]: Entering directory '/build/vorbisgain-0.37'
gcc -DHAVE_CONFIG_H -I. -I. -I.     -O2 -fsigned-char  -c `test -f 'vorbisgain.c' || echo './'`vorbisgain.c
gcc -DHAVE_CONFIG_H -I. -I. -I.     -O2 -fsigned-char  -c `test -f 'vorbis.c' || echo './'`vorbis.c
gcc -DHAVE_CONFIG_H -I. -I. -I.     -O2 -fsigned-char  -c `test -f 'recurse.c' || echo './'`recurse.c
gcc -DHAVE_CONFIG_H -I. -I. -I.     -O2 -fsigned-char  -c `test -f 'misc.c' || echo './'`misc.c
misc.c: In function 'get_console_size':
misc.c:268:9: error: implicit declaration of function 'isatty' [-Wimplicit-function-declaration]
  268 |     if (isatty(fd))
      |         ^~~~~~
make[1]: *** [Makefile:256: misc.o] Error 1
make[1]: Leaving directory '/build/vorbisgain-0.37'
make: *** [Makefile:167: all] Error 2
```

<details>

--- a/pkgs/by-name/vo/vorbisgain/fprintf.patch
+++ b/pkgs/by-name/vo/vorbisgain/fprintf.patch
@@ -1,0 +1,12 @@
+--- vorbisgain-0.37/misc.c	2004-01-03 16:17:28.000000000 -0500
++++ vorbisgain-0.37-patched/misc.c	2025-01-07 20:51:12.989521816 -0500
+@@ -56,8 +56,7 @@
+     vfprintf(stderr, message, args);
+     va_end(args);
+ 
+-    fprintf(stderr, strerror(err_num));
+-    fprintf(stderr, "\n");
++    fprintf(stderr, "%s\n", strerror(err_num));
+ }
+ 
+ 

--- a/pkgs/by-name/vo/vorbisgain/isatty.patch
+++ b/pkgs/by-name/vo/vorbisgain/isatty.patch
@@ -1,0 +1,10 @@
+--- vorbisgain-0.37/misc.c	2004-01-03 16:17:28.000000000 -0500
++++ vorbisgain-0.37-patched/misc.c	2025-01-07 20:35:13.056221211 -0500
+@@ -23,6 +23,7 @@
+ #else /* WIN32 */
+ #include <errno.h>
+ #include <ctype.h>
++#include <unistd.h>
+ 
+ #ifndef DISABLE_WINSIZE
+ 

--- a/pkgs/by-name/vo/vorbisgain/package.nix
+++ b/pkgs/by-name/vo/vorbisgain/package.nix
@@ -22,11 +22,6 @@ stdenv.mkDerivation rec {
     libvorbis
   ];
 
-  patchPhase = ''
-    chmod -v +x configure
-    configureFlags="--mandir=$out/share/man"
-  '';
-
   meta = with lib; {
     homepage = "https://sjeng.org/vorbisgain.html";
     description = "Utility that corrects the volume of an Ogg Vorbis file to a predefined standardized loudness";

--- a/pkgs/by-name/vo/vorbisgain/package.nix
+++ b/pkgs/by-name/vo/vorbisgain/package.nix
@@ -17,9 +17,8 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./isatty.patch
+    ./fprintf.patch
   ];
-
-  hardeningDisable = [ "format" ];
 
   buildInputs = [
     libogg

--- a/pkgs/by-name/vo/vorbisgain/package.nix
+++ b/pkgs/by-name/vo/vorbisgain/package.nix
@@ -15,6 +15,10 @@ stdenv.mkDerivation rec {
     sha256 = "1v1h6mhnckmvvn7345hzi9abn5z282g4lyyl4nnbqwnrr98v0vfx";
   };
 
+  patches = [
+    ./isatty.patch
+  ];
+
   hardeningDisable = [ "format" ];
 
   buildInputs = [

